add mpe_client_command with the first function: sing_message

astroseger opened the pull request:
The first function for mpe-client functionality: sign_message.
This function print signature for the claim message (see MultiPartyEscrow.sol in https://github.com/singnet/platform-contracts for details).

example: 
 snet mpe-client sign_message 0x5c7a4290f6f8ff64c69eeffdfafc8644a4ec3a4e 0 0 42  

(parameters : mpe_address, channel_id, nonce, amount)

vsbogd commented on snet_cli/mpe_client_command.py:
```diff
@@ -0,0 +1,23 @@
+class MPEClientCommand(BlockchainCommand):
+    def sign_message(self):
+        # here it is ok to accept address without checksum
+        mpe_address = self.w3.toChecksumAddress(self.args.mpe_address)
```
I agree that if we are publishing this command line interface then meaningful error description makes sense. There are two alternatives: use ```self._error``` to exit immediately or wrap and re-raise an exception.

I would say that in this case both alternatives are equivalent, but as first is used broadly in existing code may be we should stick to this way.

astroseger replied:
Vitaliy's solution is good in the situation when we need to add additional information to the error message. I think in this situation it is ok to catch exception on the higher level and print it. So I propose to add it as a separate pull request (I'm doing it right now).

vsbogd replied:
To be more clear I have proposed using ```self._error``` in this situration.

astroseger commented:
1. I've decided to add here all mpe-client functionality which we need right now.  
2. I've adopted the following strategy for error handling: catch exception, print/add additional information, re-raise exception or call self._error() directly. It is one of possibility proposed by Vitaliy.
I've added try/except at the highest level (__init__.py).  But I've added possibility (parameter --print-traceback), to print full traceback.

So now we don't need to use try/except everywhere (which makes our code ugly), if we have nothing to add to the error message then we can rely on catching exception at the highest level. Moreover if we want to debug we can use "snet --print-traceback ...." and see all traceback.

3. I will put example of using mpe-client functionality here: https://github.com/singnet/wiki/tree/master/multiPartyEscrowContract

vsbogd commented on snet_cli/mpe_client_command.py:
```diff
@@ -0,0 +1,186 @@
+                    ("snet-payment-channel-id",            str(self.args.channel_id)  ), 
+                    ("snet-payment-channel-nonce",         str(self.args.nonce)       ), 
+                    ("snet-payment-channel-amount",        str(self.args.amount)      ),
+                    ("snet-payment-channel-signature-bin", base64.b64encode(signature))]
```
As far as I understand specification you don't need encode it in base64 - gRPC library will do it for you.
